### PR TITLE
fix(UserList): Pass the desired search string when clearing filter.

### LINKED ISSUE #860: Admin: "clear search" in the users list should work on first click.
## Observed behavior (please include a screenshot if possible)

In the users list in the admin section, if a filter on user names is applied and you click on the "clear filter" button, the users list flickers but is not updated to show all users. You have to click "refresh" for the list to correctly update.

## Expected behavior

The list should show all users when clicking "clear search".

## Steps to reproduce the problem

In the users list in the admin section:
1. Enter a filter string and click "refresh"
2. Observe that the user list is filtered accordingly
3. Click "clear filter"
4. Observe that the list flickers but appears as if the filter was still applied.

## Any special notes on configuration used

NA

## Version of datatools-ui and datatools-server if applicable (exact commit hash or branch name)

Current dev branches

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Fix #860

### CODE REVIEW

File: lib/admin/components/UserList.js [hunk 1]
```diff
@@ -95,9 +97,15 @@ class UserList extends Component<Props, State> {
...
+    // Note: unable to inline because _userSearch is passed an HTML event object
+    // that we don't use.
+    this._performSearch()
+  }
+
+  _performSearch = (text?: string) => {
     const {fetchUsers, setUserPage, setUserQueryString} = this.props
     setUserPage(0)
-    setUserQueryString(this.state.searchText || '')
+    setUserQueryString(text !== undefined ? text : (this.state.searchText || ''))
```
Comment (daniel-heppner-ibigroup) [inline lib/admin/components/UserList.js:110]: Would `text || this.state.searchText || ''` not work here?

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Oops, forgot to submit my review. LGTM

File: lib/admin/components/UserList.js (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/admin/components/UserList.js:110]: Unfortunately, with that syntax, the search does not clear when you click "clear search". I added a comment regarding that in https://github.com/ibi-group/datatools-ui/pull/861/commits/12019e2aecbc02c89c86f22ef697ed9524a9751b.

**Review by miles-grant-ibigroup — APPROVED** [review]
Well done figuring out how to resolve this! Works great.

**Pull request merged**